salt/salt.spec

```diff
@@ -1101,6 +1100,18 @@ install -Dpm 0640 conf/suse/standalone-formulas-configuration.conf %{buildroot}%
 %fdupes %{buildroot}%{python3_sitelib}
 %endif
 
+%python3_fix_shebang
```

Comment:
We use this exact spec file for building different the "salt" package for different OSes, including old SLE15 versions and CentOS8/RES8, as you can see here:

https://build.opensuse.org/package/show/systemsmanagement:saltstack:products:testing/salt

This makes changes on this spec file a little bit tricky, as i.a. `%python3_fix_shebang` and some other macros are not available for all those OSes.

I would recommend to isolate your changes under the scope of `%if %{suse_version} >= 1600`, to prevent affecting the other OSes.

Reply:
I would much prefer some kind of shim, that would be more robust:

```rpm
# Compat stub for python3_fix_shebang_path
%{?!python3_fix_shebang_path:%define python3_fix_shebang_path(+abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789_-=) \\\
    myargs="%{**}" \
    for f in ${myargs}; do \
        [ -f "$f" ] && sed -i -e "1s@#\\!.*python.*@#\\!$(realpath %__python3)@" $f \
    done
    }
```